ODIN_II/SRC/netlist_utils.cpp

```diff
@@ -443,14 +449,29 @@ void combine_nets(nnet_t* output_net, nnet_t* input_net, netlist_t* netlist) {
  *-------------------------------------------------------------------------------------------*/
 void join_nets(nnet_t* join_to_net, nnet_t* other_net) {
     if (join_to_net == other_net) {
-        if ((join_to_net->driver_pin) && (join_to_net->driver_pin->node != NULL) && join_to_net->driver_pin->node->related_ast_node != NULL)
-            error_message(NETLIST, join_to_net->driver_pin->node->related_ast_node->line_number, join_to_net->driver_pin->node->related_ast_node->file_number, "%s", "This is a combinational loop\n");
-        else
-            error_message(NETLIST, -1, -1, "%s", "This is a combinational loop\n");
-        if ((join_to_net->fanout_pins[0] != NULL) && (join_to_net->fanout_pins[0]->node != NULL) && join_to_net->fanout_pins[0]->node->related_ast_node != NULL)
-            error_message(NETLIST, join_to_net->fanout_pins[0]->node->related_ast_node->line_number, join_to_net->fanout_pins[0]->node->related_ast_node->file_number, "%s", "This is a combinational loop with more info\n");
-        else
-            error_message(NETLIST, -1, -1, "%s", "Same error - This is a combinational loop\n");
+        bool has_error = false;
```

Comment:
I believe that join_to_net == other_net is the error, the fanout would never be reached in the previous version anyway, we are simply printing to allow the user to know whats going on, at which point we can just exit.

all that to say that the boolean is unecessary, this whole section is simply to tell the user